lib/pacemaker/pcmk_sched_promotable.c

```diff
@@ -753,8 +755,13 @@ pcmk__set_instance_roles(pe_resource_t *rsc, pe_working_set_t *data_set)
         score2char_stack(child_rsc->sort_index, score, len);
 
         chosen = child_rsc->fns->location(child_rsc, NULL, FALSE);
-        if (pcmk_is_set(data_set->flags, pe_flag_show_scores)) {
-            if (pcmk_is_set(data_set->flags, pe_flag_stdout)) {
+        if (pcmk_is_set(data_set->flags, pe_flag_show_scores) && !pcmk__is_daemon) {
+            if (data_set->priv != NULL) {
+                pcmk__output_t *out = data_set->priv;
+                out->info(out, "%s promotion score on %s: %s",
+                          child_rsc->id,
+                          (chosen? chosen->details->uname : "none"), score);
+            } else {
```

Comment:
I don't think we need to keep the else block here or in similar usages

Reply:
Fixed.